returnn/config.py

```diff
@@ -121,7 +121,6 @@ def __init__(
                 self.python_epilog_hash = python_epilog
         self.sort_config = sort_config
         self.pprint_kwargs = pprint_kwargs or {}
-        self.pprint_kwargs.setdefault("sort_dicts", sort_config)
```

Comment:
I would still set it for Python >=3.8.

Reply:
Or, we could also maybe be more generic:
```python
if sys.version_info[:2] <= (3, 8):
    self.pprint_kwargs.pop("sort_dicts", None)  # only supported since Python >=3.8
```
So then Sis configs works for different Python versions.

Reply:
I would rather not introduce special behavior based on the python version used to run the recipe so that all configs and parameters are the same for all users.

Also, in my opinion, sorting the config and sorting dicts within the config are two reasonably different things so that having two separate, independent config parameters makes sense.

Reply:
> Also, in my opinion, sorting the config and sorting dicts within the config are two reasonably different things so that having two separate, independent config parameters makes sense.

I disagree. I don't see any reason that you would want to sort only one of them. As you are not using this option anyway, I think we should ask the opinion of people who will use this option.

It should not be so complicated. It's not nice to have to set two options for sth which should even be the default (in my opinion).

Also, we should really not care too much about older Python versions. This is just a workaround to still support it. We could also maybe just drop Python <=3.7 support and then clean this up.

Reply:
Python 3.6.9 is the default python that comes with ubuntu 18.04 which is still what we use at AppTek (cluster + test pipeline), which is currently broken due to that change.

Reply:
So why not just use my proposed change? It should anyway not have an impact on the result. And if you anyway don't use it, it will not change anything for you.